src/Android/Xamarin.Android/Samples/Data/AddShapefile/AddShapefile.cs

```diff
@@ -0,0 +1,88 @@
+// Copyright 2017 Esri.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at: http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific
+// language governing permissions and limitations under the License.
+
+using Android.App;
+using Android.OS;
+using Android.Widget;
+using ArcGISRuntimeXamarin.Managers;
+using Esri.ArcGISRuntime.Data;
+using Esri.ArcGISRuntime.Mapping;
+using Esri.ArcGISRuntime.UI.Controls;
+using System.IO;
+
+namespace ArcGISRuntimeXamarin.Samples.AddShapefile
+{
+    [Activity]
+    public class AddShapefile : Activity
+    {
+        private MapView _myMapView;
+
+        protected override void OnCreate(Bundle bundle)
+        {
+            base.OnCreate(bundle);
+
+            Title = "Add shapefile";
+            CreateLayout();
+
+            // Download (if necessary) and add a local shapefile dataset to the map
+            Initialize();
+        }
+
+        private async void Initialize()
+        {
+            // Create a new map to display in the map view with a streets basemap
+            _myMapView.Map = new Map(Basemap.CreateStreets());
+
+            // The shapefile will be downloaded from ArcGIS Online
```

Comment:
There's a new process for offline data documented [here](https://github.com/Esri/arcgis-runtime-samples-dotnet/wiki/Offline-Data). I'm just realizing now that I never told anyone... 

Basically, the data manager code needs to go into a separate function with region tags. That makes it easier for the tooling to replace it with more appropriate code when injected into the API ref.